src/promptflow/promptflow/_sdk/_telemetry/activity.py

```diff
@@ -233,10 +242,17 @@ def wrapper(self, *args, **kwargs):
 
             logger = get_telemetry_logger()
 
-            custom_dimensions.update(extract_telemetry_info(self))
+            extra_custom_dimensions, bonded_user_agent = extract_telemetry_info(self)
+            custom_dimensions.update(extra_custom_dimensions)
             # update activity name according to kwargs.
             _activity_name = update_activity_name(activity_name, kwargs=kwargs)
-            with log_activity(logger, _activity_name, activity_type, custom_dimensions):
+            with log_activity(
+                logger=logger,
+                activity_name=_activity_name,
+                activity_type=activity_type,
+                custom_dimensions=custom_dimensions,
+                user_agent=bonded_user_agent,
```

Comment:
Can we not add this **user_agent** parameter? Directly obtain **bonded_user_agent** within **log_activity** function?